src/main/java/dev/aziz/grocerystore/handlers/PriceHandler.java

```diff
@@ -1,27 +1,64 @@
 package dev.aziz.grocerystore.handlers;
 
 import dev.aziz.grocerystore.entities.BasketItem;
+import dev.aziz.grocerystore.entities.UserPromotion;
+import lombok.extern.slf4j.Slf4j;
 
 import java.math.BigDecimal;
 import java.math.BigInteger;
+import java.time.Instant;
 import java.util.List;
 
+@Slf4j
 public class PriceHandler {
 
+    private final UserPromotion userPromotion;
+
     private List<BasketItem> basketItems;
 
-    public PriceHandler(List<BasketItem> basketItems) {
+    public PriceHandler(UserPromotion userPromotion, List<BasketItem> basketItems) {
+        this.userPromotion = userPromotion;
         this.basketItems = basketItems;
     }
 
     public BigDecimal computePrice() {
+
         BigDecimal totalPrice = new BigDecimal(BigInteger.ZERO);
-        for (BasketItem basketItem : basketItems) {
-            totalPrice = totalPrice.add(
-                    basketItem.getItem().getPrice().multiply(BigDecimal.valueOf(basketItem.getAmount()))
-            );
+
+        if (userPromotion.getPromotionConfig() != null) {
```

Comment:
You must iterate over all the items of the basket, and for each item, check if there is a promotion active